src/Enzo/particle/formation/EnzoMethodStarMakerSTARSS.cpp

```diff
@@ -229,32 +229,37 @@ void EnzoMethodStarMakerSTARSS::compute ( Block *block) throw()
 
   compute_temperature.compute(enzo_block);
 
+  const GrackleChemistryData * grackle_chem = enzo::grackle_chemistry();
+  const int primordial_chemistry = (grackle_chem == nullptr) ?
+    0 : grackle_chem->get<int>("primordial_chemistry");
 
-  double mu = static_cast<double>(enzo::fluid_props()->mol_weight());
+  const double dflt_mu = static_cast<double>(enzo::fluid_props()->mol_weight());
   // iterate over all cells (not including ghost zones)
   for (int iz=gz; iz<nz+gz; iz++){
     for (int iy=gy; iy<ny+gy; iy++){
       for (int ix=gx; ix<nx+gx; ix++){
 
         int i = INDEX(ix,iy,iz,mx,my);//ix + mx*(iy + my*iz);
-     
+
+        double mu;
```

Comment:
I'm not sure if it's worth adding the influence of the metals on mu here, but I will mention it just in case.

Reply:
I'm not sure if it's worth adding here either. I'm a little hesitant to make any real changes here since I've never used this section of the codebase... 

I added a comment about this showing the necessary code to do that. 

I'll file an issue to ask what people want to do. That can be addressed in a separate short PR that I'm happy to make (long-term, we should either have a comment here explaining that exclusion of metal density is a conscious choice or we should fix it).

**While I was here**, I also fixed a bug that in the default that I had previously introduced. I also some comments explaining what's going on in the calculation.